src/build/build.ts

```diff
@@ -77,9 +75,9 @@ export function build(options: BuildOptions, config: ProjectConfig): Promise<any
     logger.debug(`Reading source files...`);
     let sourcesStream = polymerProject.sources()
       .pipe(polymerProject.splitHtml())
-      .pipe(gulpif(/\.js$/, uglify(optimizeOptions.js)))
-      .pipe(gulpif(/\.css$/, cssSlam(optimizeOptions.css)))
-      .pipe(gulpif(/\.html$/, htmlmin(optimizeOptions.html)))
+      .pipe(gulpif(/\.js$/, new JSOptimizeStream(optimizeOptions.js)))
```

Comment:
Also check out https://github.com/hughsk/vinyl-transform & [the relevant blog post it mentions](http://blog.overzealous.com/post/74121048393/why-you-shouldnt-create-a-gulp-plugin-or-how-to). This should be a fine solution for our users (and the best part is it already exists :)